agent/src/main/java/com/teamscale/jacoco/agent/options/TeamscaleProxyOptions.java

```diff
@@ -0,0 +1,115 @@
+package com.teamscale.jacoco.agent.options;
+
+import com.teamscale.client.ProxySystemProperties;
+import com.teamscale.client.TeamscaleProxySystemProperties;
+import com.teamscale.report.util.ILogger;
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Parses agent command line options related to the proxy settings.
+ */
+public class TeamscaleProxyOptions {
+
+	private final ILogger logger;
+
+	/** The host of the proxy server.  */
+	/* package */ String proxyHost;
+
+	/** The port of the proxy server. */
+	/* package */ int proxyPort;
+
+	/** The password for the proxy user. */
+	/* package */ String proxyPassword;
+
+	/** A path to the file that contains the password for the proxy authentication. */
+	/* package */ Path proxyPasswordPath;
+
+	/** The username of the proxy user. */
+	/* package */ String proxyUser;
+
+	private final ProxySystemProperties.Protocol protocol;
+
+	/** Constructor. */
+	public TeamscaleProxyOptions(ProxySystemProperties.Protocol protocol, ILogger logger) {
+		this.protocol = protocol;
+		this.logger = logger;
+		ProxySystemProperties proxySystemProperties = new ProxySystemProperties(protocol);
+		proxyHost = proxySystemProperties.getProxyHost();
+		try {
+			proxyPort = proxySystemProperties.getProxyPort();
+		} catch (ProxySystemProperties.IncorrectPortFormatException e) {
+			proxyPort = -1;
+			logger.warn(e.getMessage());
+		}
+		proxyUser = proxySystemProperties.getProxyUser();
+		proxyPassword = proxySystemProperties.getProxyPassword();
+	}
+
+	/**
+	 * Handles all command-line options prefixed with 'proxy-'
+	 *
+	 * @return true if it has successfully processed the given option.
+	 */
+	public boolean handleTeamscaleProxyOptions(String key, String value) throws AgentOptionParseException {
+			if (String.format("proxy-%s-host", protocol).equals(key)){
+				proxyHost = value;
+				return true;
+			}
+		String proxyPortOption = "proxy-%s-port";
+		if (String.format(proxyPortOption, protocol).equals(key)) {
+				try {
+					proxyPort = Integer.parseInt(value);
+				} catch (NumberFormatException e) {
+					throw new AgentOptionParseException("Could not parse proxy port \"" + value +
+							"\" set via \"" + proxyPortOption + "\"", e);
+				}
+				return true;
+			}
+			if (String.format("proxy-%s-user", protocol).equals(key)) {
+				proxyUser = value;
+				return true;
+			}
+			if (String.format("proxy-%s-password", protocol).equals(key)) {
+				proxyPassword = value;
+				return true;
+			}
+			return false;
+	}
+
+	/** Stores the teamscale-specific proxy settings as system properties to make them always available. */
+	public void putTeamscaleProxyOptionsIntoSystemProperties() {
+		TeamscaleProxySystemProperties teamscaleProxySystemProperties = new TeamscaleProxySystemProperties(protocol);
+		if (proxyHost != null && !proxyHost.isEmpty()) {
```

Comment:
We can just use StringUtils.isEmpty() for that
```suggestion
		if (!StringUtils.isEmpty(proxyHost)) {
```